x-pack/plugins/alerting/server/plugin.test.ts

```diff
@@ -242,6 +243,7 @@ describe('Alerting Plugin', () => {
               features: featuresPluginMock.createSetup(),
               unifiedSearch: autocompletePluginMock.createSetupContract(),
               ...(useDataStreamForAlerts ? { serverless: {} } : {}),
+              serverless: serverlessPluginMock.createSetupContract(),
```

Comment:
I don't think this is going to work right, or is at least confusing.  The new line sets a `serverless` property, but so does the line above, optionally.  In addition, the line above only sets the value when `useDataStreamForAlerts` is true.  With the new code, it will always be set, unless for some reason the optional one above it "wins" (which I don't think it will).  

I was surprised no tests failed because of this, but this is testing the plugin `start()` method, which doesn't currently have dependencies on the `serverless` plugin.  If it did, I'd guess we'd get some failures.

The intention here is that we are testing "with" and "without" the serverless plugin being available, which is determined in the tests via the `useDataStreamForAlerts` value.  We are basically not testing the case of the `serverless` plugin **NOT** being available, only testing it **BEING** available.

I think the code should be changed to the version on line 76 now.  

Same is true for the similiar changes below ...